lib/ext.js

```diff
@@ -68,71 +61,65 @@ module.exports = class Ext {
     }
 
     name(arg) {
-
         return this.config.meta[arg].name;
     }
 
     assignIfActive(meta, name, value) {
-
         if (this.config.meta[name].active) {
             meta[this.name(name)] = value;
         }
     }
 
     onPreHandler(request, h) {
-
         // If the route does not match, just skip this part
-        if (this.isValidRoute(request)) {
-
-            const routeDefaults = this.getRouteOptions(request).defaults || {};
-            const pagination = this.getPagination(request);
-            request.query[this.config.query.pagination.name] = pagination;
+        if (!this.isValidRoute(request)) {
+            return h.continue;
+        }
 
-            if (pagination === false) {
-                return h.continue;
-            }
+        const routeDefaults = this.getRouteOptions(request).defaults || {};
+        const pagination = this.getPagination(request);
+        request.query[this.config.query.pagination.name] = pagination;
 
-            const page = routeDefaults.page || this.config.query.page.default;
-            const limit = routeDefaults.limit || this.config.query.limit.default;
+        if (pagination === false) {
+            return h.continue;
+        }
 
-            const setParam = (arg, defaultValue) => {
+        const page = routeDefaults.page || this.config.query.page.default;
+        const limit = routeDefaults.limit || this.config.query.limit.default;
 
-                const name = this.name(arg);
-                let value = null;
+        const setParam = (arg, defaultValue) => {
+            const name = this.name(arg);
+            let value = null;
 
-                if (request.query[name]) {
-                    value = parseInt(request.query[name]);
+            if (request.query[name] != null) {
```

Comment:
```suggestion
            if (request.query[name] !== null) {
```

Reply:
Used Utils.isNil which achieves the same intended check ("not null or undefined").